providers/file.go

```diff
@@ -0,0 +1,66 @@
+package providers
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// File represents a provider
+type File struct {
+	*BaseProvider
+}
+
+// FileConfig stores a file configuration
+type FileConfig struct {
+	Path    string      `yaml:"path"`
+	Content string      `yaml:"content"`
+	Group   string      `yaml:"group,omitempty"`
+	Mode    os.FileMode `yaml:"mode,omitempty"`
+	Owner   string      `yaml:"owner,omitempty"`
+}
+
+// GetName returns the provider's name
+func (f *File) GetName() string {
+	return "File"
+}
+
+// Unseal unseals a secret from the backend and add it to the config file
+func (f *File) Unseal(cred string) (err error) {
+	var secret FileConfig
+
+	err = f.backend.GetSecret(f.inputPath+"/"+cred, &secret)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve credentials: %s", err)
+	}
+
+	outputFile := os.ExpandEnv(secret.Path)
```

Comment:
[os.ExpandEnv()](https://golang.org/pkg/os/#ExpandEnv) will translate environment variables, "\~" will not be interpreted. You could use the function [strings.Replace()](https://golang.org/pkg/strings/#Replace) to replace "\~" by "$HOME" or use the gopass function [fsutil.CleanPath()](https://godoc.org/github.com/gopasspw/gopass/pkg/fsutil).